python/lsst/daf/butler/registries/oracleRegistry.py

```diff
@@ -48,7 +49,7 @@ def _merge(merge, compiler, **kw):
     pkColumns = [col.name for col in table.primary_key]
     nonPkColumns = [col for col in allColumns if col not in pkColumns]
 
-    selectColumns = [bindparam(col).label(col) for col in allColumns]
+    selectColumns = [bindparam(col.name, type_=col.type).label(col.name) for col in table.columns]
```

Comment:
For future code readers, a comment about why the type_ is needed would probably be helpful.